### ISSUE TITLE
Using training or validation data for the MMN network updates?

### ISSUE THREAD

**Issue by laubeke** [microsoft/AutoML#23]
Hi, thanks for the nice work and code!

According to the paper, the MMN network is updated based on the super-networks on the validation data (see https://papers.nips.cc/paper/2020/file/d072677d210ac4c03ba046120f0802ec-Paper.pdf close to the end of page 4).
On the other hand, in the code, the MMN network is updated in https://github.com/microsoft/Cream/blob/main/lib/core/train.py#L42 and uses training data.

Am I missing something?

**Comment by Z7zuqer** [comment]
Hi,

Thanks for your interest in our project!

Different slices of current batch are used in the meta network update process.

Training: https://github.com/microsoft/Cream/blob/e0de68affb0827b2a0108e7afa633dbc97b233f0/lib/models/MetaMatchingNetwork.py#L50

Validation: https://github.com/microsoft/Cream/blob/e0de68affb0827b2a0108e7afa633dbc97b233f0/lib/models/MetaMatchingNetwork.py#L86

Best,
Hao.

**Comment by laubeke** [comment]
Hi,

thanks for the quick response!
I was uncertain whether an entirely different validation set is used, or whether training data can be used both for matching and MMN training, thanks for the clarification.

Best,
Kevin

**Comment by Z7zuqer** [comment]
Hi,

Thanks for your interest in our project!

'Validation' used in the update of MMN is not from ImageNet Val, instead images in 'Validation' are sampled from ImageNet Train. To summarize, in the process of updating MMN, we split ImageNet TrainSet into *Train* and *Val*, only for MMN. ImageNet Val images are only used in the process of calculating validation loss of supernet.

Best,
Hao.
